roles/pool/src/main.rs

```diff
@@ -22,68 +27,147 @@ pub type Message = PoolMessages<'static>;
 pub type StdFrame = StandardSv2Frame<Message>;
 pub type EitherFrame = StandardEitherFrame<Message>;
 
-const BLOCK_REWARD: u64 = 5_000_000_000;
-
-//const COINBASE_ADD_SZIE: u32 = 100;
+const BLOCK_REWARD: u64 = 625_000_000;
 
-pub fn get_coinbase_output(config: &Configuration) -> Vec<TxOut> {
-    config
+pub fn get_coinbase_output(config: &Configuration) -> Result<Vec<TxOut>, OutputScriptError> {
+    let result = config
         .coinbase_outputs
         .iter()
-        .map(|pub_key_wrapper| {
-            let hashed = pub_key_wrapper.pub_key.pubkey_hash();
-            TxOut {
-                // value will be updated by the addition of `ChannelFactory::split_outputs()` in PR #422
+        .map(|coinbase_output| {
+            coinbase_output.try_into().map(|output_script| TxOut {
                 value: crate::BLOCK_REWARD,
-                script_pubkey: Script::new_p2pkh(&hashed),
-            }
+                script_pubkey: output_script,
+            })
         })
-        .collect()
+        .collect::<Result<Vec<TxOut>, OutputScriptError>>();
+
+    if result.is_ok() && result.as_ref().unwrap().is_empty() {
+        Err(OutputScriptError::EmptyCoinbaseOutputs(
+            "Empty coinbase outputs".to_string(),
+        ))
+    } else {
+        result
+    }
 }
 
-use tokio::select;
+impl TryFrom<&CoinbaseOutput> for Script {
+    type Error = OutputScriptError;
 
-use crate::status::Status;
+    fn try_from(value: &CoinbaseOutput) -> Result<Self, Self::Error> {
+        match value.output_script_type.as_str() {
+            "P2PK" => {
+                if is_public_key(&value.output_script_value) {
```

Comment:
Same as above no need to check if the value is valid cause we do that later

Reply:
Also valid for all the other branches

Reply:
Also maybe this logic and the one above about trasforming a toml value into a TxOut should stay in roles-logic-sv2 utils

Reply:
> Also maybe this logic and the one above about trasforming a CoinbaseOutput into a TxOut should stay in roles-logic-sv2 utils

I agree.
I will do this after the PR which will be opened by @vincenzopalazzo to create a stratum-common to fix the rust-bitcoin thing. Because I tried to move this logic in `roles-logic-sv2/utils.rs` but I messed up with rust-bitcoin version which doesn't support the `PublicKey::fromstr()` method anymore.